lib/Service/FileService.php

```diff
@@ -1719,24 +1792,72 @@ public function getFile(ObjectEntity | string $object, string $filePath): ?File
      */
     public function publishFile(ObjectEntity | string $object, string $filePath): File
     {
-        // If string ID provided, try to find the object entity
-        if (is_string($object) === true) {
-            $object = $this->objectEntityMapper->find($object);
-        }
+        return $this->executeWithFileUserContext(function () use ($object, $filePath): File {
```

Comment:
This is some weird code 🤔
Return immediately an existing function with a new function call?

Reply:
It is an unified way to execute functions that should be run in a different user throughout the service, by feeding it to another function.
We have noted that switching users should disappear as soon as possible, but this way it is always going through the same code.